launchable/commands/record/build.py

```diff
@@ -248,3 +248,46 @@ def build(ctx: click.core.Context, build_name: str, source: List[str], max_days:
             ))
 
     write_build(build_name)
+
+
+def _get_branch_name(repo_dist: str) -> Optional[str]:
```

Comment:
[question]
When does this return None?
I see the code may return "", but cannot determine None.

Reply:
I know your opinion  but if I changed `Optional[str]` to `str` , these errors will happen

```
$ pipenv run type
launchable/commands/record/build.py:260: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
launchable/commands/record/build.py:262: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
launchable/commands/record/build.py:267: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
launchable/commands/record/build.py:269: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
launchable/commands/record/build.py:274: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
```

Reply:
Which is better 1. Keep Optional[str] 2. Change to `return os.environ.get(JENKINS_GIT_BRANCH_KEY, "")`

Reply:
Since you are checking the env var existence in the code, then I think you can just return `os.environ["KEY"]`.
This should return `str` for value of env var dict.

I suppose it is better to avoid None (or nil in Go, Null in Java) than accept None if possible.

Reply:
Thank you! I fixed it https://github.com/launchableinc/cli/pull/570/commits/e448da1b6ae1a4b6287ca9eb951c2f774ca50c1d